draft-ietf-dnsop-svcb-https.md

```diff
@@ -1158,8 +1161,10 @@ the origin, not the TargetName.
 ## HTTP Strict Transport Security {#hsts}
 
 By publishing a usable HTTPS RR, the server operator indicates that all
-useful HTTP resources on that origin are reachable over HTTPS, similar to
-HTTP Strict Transport Security {{HSTS}}.
+useful HTTP resources on this Host ({{!I-D.draft-ietf-httpbis-semantics,
+Section 7.2}}) are reachable over HTTPS, similar to HTTP Strict Transport
```

Comment:
```suggestion
origins with an "http" scheme governed by this host ({{!I-D.draft-ietf-httpbis-semantics,
Section 4.3.2}}) have an equivalent origin with an "https" scheme ({{!I-D.draft-ietf-httpbis-semantics,
Section 4.3.3}}), similar to HTTP Strict Transport
```

I think Mark's point is that we're making a statement about mapping between one set of origins and another, not individual resources or a single origin.  The origin transformation rules imported from HSTS below reflect this.

Reply:
I'm sure your terminology is an improvement, but I think we need to tweak this.

My concern with this phrasing is that "equivalent origin" suggests that the resources on both origins should be the same, but that's not actually what we want.  Ideally, the HTTP origin will be blank except for a redirect to HTTPS.  That's the rationale for the current language about "all useful HTTP resources".

Can you think of a way to represent this idea without implying the that origins have the same content?

Reply:
"By publishing a usable HTTPS RR, the server operator indicates that all origins with an "http" scheme governed by this host have HTTPS resources have an origin with the correct "https" scheme similar to HTTP Strict Transport"

no that sucks too.   I find "equivalent" an issue as well after reading it over a few trimes.

Reply:
Seems like what we want is to communicate that you're going to end up with the same content either way.  That might be because the http:// endpoint serves content, or it might be because it sends you a 3XX telling you that you should have done https://.

If the issue is with the term "equivalent," how about "associated"?  The origins are cooperating, they will get you to the same content (though the mechanism may vary), but they may not return identical responses.

Reply:
I like associated.

Reply:
I've tried to rewrite this, borrowing more text from the HSTS RFC and some of @MikeBishop's language.  It now reads

>An HTTPS RR directs the client to communicate with this host only over a
> secure transport, similar to HTTP Strict Transport Security {{HSTS}}.

...

> If the "http" origin exists and contains important resources that have no
> equivalent at the "https" origin, the operator MUST NOT publish an HTTPS RR.